ddl/ingest/backend.go

```diff
@@ -76,26 +83,26 @@ func (bc *BackendContext) FinishImport(indexID int64, unique bool, tbl table.Tab
 const importThreshold = 0.85
 
 // Flush checks the disk quota and imports the current key-values in engine to the storage.
-func (bc *BackendContext) Flush(indexID int64) error {
+func (bc *BackendContext) Flush(indexID int64) (imported bool, err error) {
 	ei, exist := bc.EngMgr.Load(indexID)
 	if !exist {
 		logutil.BgLogger().Error(LitErrGetEngineFail, zap.Int64("index ID", indexID))
-		return dbterror.ErrIngestFailed.FastGenByArgs("ingest engine not found")
+		return false, dbterror.ErrIngestFailed.FastGenByArgs("ingest engine not found")
 	}
 
-	err := bc.diskRoot.UpdateUsageAndQuota()
+	err = bc.diskRoot.UpdateUsageAndQuota()
 	if err != nil {
 		logutil.BgLogger().Error(LitErrUpdateDiskStats, zap.Int64("index ID", indexID))
-		return err
+		return false, err
 	}
 
+	err = ei.Flush()
```

Comment:
Put ei.Flush() outside of the block may cause backfilling.go:1054 line, here will call flush for each round maybe not necessary or performance impact.

Reply:
This function is only invoked before the add index worker handling every 1024 regions.